UnityAds/UnityAdsInterstitialCustomEvent.m

```diff
@@ -126,84 +125,61 @@ - (BOOL)enableAutomaticImpressionAndClickTracking
     return NO;
 }
 
-#pragma mark - UnityRouterDelegate
+#pragma mark - UnityAdsLoadDelegate Methods
 
-- (void)unityAdsReady:(NSString *)placementId
-{
-    // No-op
+- (void)unityAdsAdLoaded:(NSString *)placementId {
+    [self.delegate fullscreenAdAdapterDidLoadAd:self];
+    MPLogAdEvent([MPLogEvent adLoadSuccessForAdapter:NSStringFromClass(self.class)], [self getAdNetworkId]);
 }
 
-/// This method only handles show-related errors. Load-related errors are handled by unityAdsAdFailedToLoad.
-- (void)unityAdsDidError:(UnityAdsError)error withMessage:(NSString *)message
-{
-    [UnityAds removeDelegate:self];
-    
-    if (error == kUnityAdsErrorShowError) {
-      NSError *showError= [self createErrorWith:@"Unity Ads failed to show interstitial"
-                                      andReason:message
-                                  andSuggestion:@""];
-        MPLogAdEvent([MPLogEvent adShowFailedForAdapter:NSStringFromClass(self.class) error:showError], [self getAdNetworkId]);
-        [self.delegate fullscreenAdAdapter:self didFailToShowAdWithError:showError];
-    } else {
-        NSError *otherError= [self createErrorWith:@"Unity Ads interstitial failure"
-                                           andReason:message
-                                       andSuggestion:@""];
-        MPLogAdEvent([MPLogEvent adShowFailedForAdapter:NSStringFromClass(self.class) error:otherError], [self getAdNetworkId]);
-        [self.delegate fullscreenAdAdapter:self didFailToShowAdWithError:otherError];
-    }
+- (void)unityAdsAdFailedToLoad:(nonnull NSString *)placementId withError:(UnityAdsLoadError)error withMessage:(NSString *)message {
+    NSError *errorLoad = [self createErrorWith:@"Unity Ads failed to load interstitial ad"
+                                     andReason:@""
+                                 andSuggestion:@""];
```

Comment:
do we need to detail this for better formatting?

Reply:
+1, should we add the message in the `andReason` ?